{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "지역변수 전역변수 구별가능\n",
    "람다함수\n",
    "함수 호출과 반환\n",
    "csv 작업, 디렉토리 작업, 이진 파일 x\n",
    "\n",
    "과제 문제 이해하고 자유자재로 해야함!\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\FreshSider\\AppData\\Local\\Temp\\ipykernel_9796\\1130125994.py\", line 12, in displayArc\n",
      "    canvas.create_arc(10,10,WIDTH-10, HEIGHT-10, start=0,\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2811, in create_arc\n",
      "    return self._create('arc', args, kw)\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2805, in _create\n",
      "    return self.tk.getint(self.tk.call(\n",
      "_tkinter.TclError: unknown color name \"bule\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\FreshSider\\AppData\\Local\\Temp\\ipykernel_9796\\1130125994.py\", line 12, in displayArc\n",
      "    canvas.create_arc(10,10,WIDTH-10, HEIGHT-10, start=0,\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2811, in create_arc\n",
      "    return self._create('arc', args, kw)\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2805, in _create\n",
      "    return self.tk.getint(self.tk.call(\n",
      "_tkinter.TclError: unknown color name \"bule\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\FreshSider\\AppData\\Local\\Temp\\ipykernel_9796\\1130125994.py\", line 12, in displayArc\n",
      "    canvas.create_arc(10,10,WIDTH-10, HEIGHT-10, start=0,\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2811, in create_arc\n",
      "    return self._create('arc', args, kw)\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2805, in _create\n",
      "    return self.tk.getint(self.tk.call(\n",
      "_tkinter.TclError: unknown color name \"bule\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\FreshSider\\AppData\\Local\\Temp\\ipykernel_9796\\1130125994.py\", line 12, in displayArc\n",
      "    canvas.create_arc(10,10,WIDTH-10, HEIGHT-10, start=0,\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2811, in create_arc\n",
      "    return self._create('arc', args, kw)\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2805, in _create\n",
      "    return self.tk.getint(self.tk.call(\n",
      "_tkinter.TclError: unknown color name \"bule\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\FreshSider\\AppData\\Local\\Temp\\ipykernel_9796\\1130125994.py\", line 12, in displayArc\n",
      "    canvas.create_arc(10,10,WIDTH-10, HEIGHT-10, start=0,\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2811, in create_arc\n",
      "    return self._create('arc', args, kw)\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2805, in _create\n",
      "    return self.tk.getint(self.tk.call(\n",
      "_tkinter.TclError: unknown color name \"bule\"\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 1921, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"C:\\Users\\FreshSider\\AppData\\Local\\Temp\\ipykernel_9796\\1130125994.py\", line 12, in displayArc\n",
      "    canvas.create_arc(10,10,WIDTH-10, HEIGHT-10, start=0,\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2811, in create_arc\n",
      "    return self._create('arc', args, kw)\n",
      "  File \"C:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.10_3.10.3056.0_x64__qbz5n2kfra8p0\\lib\\tkinter\\__init__.py\", line 2805, in _create\n",
      "    return self.tk.getint(self.tk.call(\n",
      "_tkinter.TclError: unknown color name \"bule\"\n"
     ]
    }
   ],
   "source": [
    "from tkinter import *\n",
    "\n",
    "WIDTH = 600\n",
    "HEIGHT = 200\n",
    "\n",
    "def displayRect():\n",
    "    canvas.create_rectangle(10,10,WIDTH-10, HEIGHT-10)\n",
    "\n",
    "def displayOval():\n",
    "    canvas.create_oval(10,10,WIDTH-10, HEIGHT-10, fill=\"yellow\")\n",
    "def displayArc():\n",
    "    canvas.create_arc(10,10,WIDTH-10, HEIGHT-10, start=0,\n",
    "                    extent=120,width=10,fill='bule')\n",
    "def displayPolygon():\n",
    "    canvas.create_polygon(10,10,WIDTH-10, HEIGHT-10,200,90,300,160)\n",
    "def displayLine():\n",
    "    canvas.create_line(10,10,WIDTH-10, HEIGHT-10, fill='green')\n",
    "def clearCanvas():\n",
    "    canvas.delete(ALL)\n",
    "\n",
    "root = Tk()\n",
    "canvas = Canvas(root, width=WIDTH, height=HEIGHT, bg=\"white\")\n",
    "canvas.pack()\n",
    "frame = Frame(root)\n",
    "frame.pack()\n",
    "\n",
    "Button(frame, text=\"Rectangle\", command=displayRect).grid(row=1, column=2)\n",
    "Button(frame, text=\"Oval\", command=displayOval).grid(row=1, column=3)\n",
    "Button(frame, text=\"Arc\", command=displayArc).grid(row=1, column=5)\n",
    "Button(frame, text=\"Polygon\", command=displayPolygon).grid(row=1, column=4)\n",
    "Button(frame, text=\"Line\", command=displayLine).grid(row=1, column=1)\n",
    "Button(frame, text=\"Clear\", command=clearCanvas).grid(row=1, column=7)\n",
    "\n",
    "root.mainloop()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
